File: aspnet/AppWeb1.4/AppWeb1.4/UserIcon.ashx.vb
Imports System.Web
Imports System.IO
Imports System.Web.Services

Public Class UserIcon
    Implements System.Web.IHttpHandler
    Dim Main As New JDB()

    Sub ProcessRequest(ByVal context As HttpContext) Implements IHttpHandler.ProcessRequest
        If IsNumeric(context.Request.QueryString("i")) Then

            Dim cmd As New System.Data.SqlClient.SqlCommand
            Dim conn As New System.Data.SqlClient.SqlConnection
            conn.ConnectionString = Main.ConnStr
            conn.Open()
            cmd.Connection = conn
            cmd.CommandText = "Select User_Icon from  Users where IDNo=" & context.Request.QueryString("i")
            'context.Response.ContentType = "image/jpeg"
            'Response.Write(dr("sFileName"))
        
            Dim img As Object = cmd.ExecuteScalar
            If Not IsDBNull(img) Then
                Dim MS As MemoryStream = New MemoryStream(CType(img, Byte()))
                context.Response.ContentType = "image/jpeg"
                context.Response.BinaryWrite(MS.ToArray())

            Else
                Dim aa As New FileStream(context.Server.MapPath("img/defaulthead.png"), FileMode.Open)
                Dim buf(aa.Length - 1) As Byte
                If aa.Read(buf, 0, aa.Length) > 1 Then
                    context.Response.ContentType = "image/png"
                    context.Response.BinaryWrite(buf)
                End If
                aa.Close()
                aa.Dispose()
            End If
            conn.Close()

            'Do While byteSeq > 0
            '    context.Response.OutputStream.Write(buffer, 0, byteSeq)
            '    byteSeq = MemoryStream.Read(buffer, 0, 4096)
            'Loop
        End If

    End Sub

    ReadOnly Property IsReusable() As Boolean Implements IHttpHandler.IsReusable
        Get
            Return False
        End Get
    End Property

End Class

File: aspnet/MiniStore/UserIcon.ashx.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.IO;
using System.Web;

namespace AppPortal
{
    /// <summary>
    /// UserIcon 的摘要描述
    /// </summary>
    public class UserIcon : IHttpHandler
    {
        CommTool Comm = new CommTool();
        JDB Main = new JDB();

        public void ProcessRequest(HttpContext context)
        {
            if (Comm.IsNumeric(context.Request.QueryString["i"]))
            {
                SqlCommand cmd = new SqlCommand();
                SqlConnection conn = new SqlConnection();
                conn.ConnectionString = Main.ConnStr;
                conn.Open();
                cmd.Connection = conn;
                cmd.CommandText = "Select User_Icon from  Users where IDNo=" + context.Request.QueryString["i"];
                //object img = cmd.ExecuteScalar();              
                //if (img.GetType() !=  System.DBNull.Value.GetType())
                //{
                //    MemoryStream MS = new MemoryStream(Convert.ToByte(img));
                //    context.Response.ContentType = "image/jpeg";
                //    context.Response.BinaryWrite(MS.ToArray());
                //    MS.Close();
                //}
                //else
                //{
                //    FileStream aa = new FileStream(context.Server.MapPath("img/defaulthead.png"), FileMode.Open);
                //    byte[] buf = new byte[aa.Length];
                //    if (aa.Read(buf, 0, int.Parse(aa.Length.ToString())) > 1)
                //    {
                //        context.Response.ContentType = "image/png";
                //        context.Response.BinaryWrite(buf);
                //    }
                //    aa.Close();
                //    aa.Dispose();
                //}


                SqlDataReader dr = cmd.ExecuteReader();
                bool chk = false;
                if (dr.Read())
                {
                    if (dr["User_Icon"].GetType() != System.DBNull.Value.GetType())
                    {
                        byte[] imageData = (byte[])dr["User_Icon"];
                        MemoryStream ms = new MemoryStream(imageData, 0, imageData.Length);
                        context.Response.ContentType = "image/jpeg";
                        context.Response.BinaryWrite(ms.ToArray());
                        ms.Close();
                        chk = true;
                    }
                }

                if (chk == false)
                {
                    FileStream aa = new FileStream(context.Server.MapPath("img/defaulthead.png"), FileMode.Open);
                    byte[] buf = new byte[aa.Length];
                    if (aa.Read(buf, 0, int.Parse(aa.Length.ToString())) > 0)
                    {
                        context.Response.ContentType = "image/png";
                        context.Response.BinaryWrite(buf);
                    }
                    aa.Close();
                    aa.Dispose();
                }
                conn.Close();

            }
        }

        public bool IsReusable
        {
            get
            {
                return false;
            }
        }
    }
}